.github/workflows/test.yml

```diff
@@ -3,23 +3,31 @@ name: Build and Test
 
 on:
   push:
+    branches:
+      - main
   pull_request:
-  repository_dispatch:
-  schedule:
-     - cron: '05 5 1 * *' # <https://crontab.guru/#05_5_1_*_*> - "At 05:05 on day-of-month 1"  
+#  repository_dispatch:
+#  schedule:
+#     - cron: '05 5 1 * *' # <https://crontab.guru/#05_5_1_*_*> - "At 05:05 on day-of-month 1"
 
 
 jobs:
 
   build-and-test:
-    runs-on: windows-latest 
+    runs-on:
+#      - windows-latest
+      - self-hosted
+      - ephemeral
     
     steps:
     - name: Checkout
-      uses: actions/checkout@v2
+      uses: actions/checkout@v4
     
     - name: Setup Go environment
-      uses: actions/setup-go@v2.2.0
+      uses: actions/setup-go@v5
+      with:
+        go-version: 1.22.3
+        cache: false
     
     - name: Build
       run: go build -v ./...
```

Comment:
tbh not sure if this is going to work. It didn't run on my PR.

Reply:
<img width="1221" alt="image" src="https://github.com/customerio/go-approval-tests/assets/32811552/d89d1554-18de-4bb4-ba81-207c2518c948">

Reply:
https://github.com/customerio/go-approval-tests/actions